### DISCUSSION TITLE
linux mint: apps cant find plugin, but I can....

### DISCUSSION THREAD

**Discussion by lovatone** [streamlink/streamlink#4333] [Q&A] [answered]
Hi, 
In terminal, when i run
greg@GregsAMC:~$ streamlink https://pluto.tv/live-tv/star-trek-1 best
[cli][info] Found matching plugin pluto for URL https://pluto.tv/live-tv/star-trek-1
[cli][info] Available streams: 570k (worst), 1000k, 1500k, 2100k, 3100k (best)
[cli][info] Opening stream: 3100k (hls-pluto)
[cli][info] Starting player: /usr/bin/vlc..... well, it works great!

but running
greg@GregsAMC:~$  sudo -u nextpvr streamlink https://pluto.tv/live-tv/watchfree-star-trek-1 best
error: No plugin can handle URL: https://pluto.tv/live-tv/watchfree-star-trek-1

With my limited brain power it seems to me that the PATH is not updated or there's a permissions issue.
How can i make the plug in available to m\calls from my app? (What did i do wrong on the installation?)
thank you,
Greg

**Comment by mkbloke** [comment]
Please run each of those again with `-l debug` and post the full output for each.

**Reply by lovatone** [reply]
Thank you for your help!
Here are the results.

The NextPvr support team gave me the two command lines to test.

greg@GregsAMC:/Downloads$ streamlink https://pluto.tv/live-tv/gunsmoke best -l debug
[cli][debug] OS:         Linux-5.4.0-962201191901-generic-x86_64-with-glibc2.29
[cli][debug] Python:     3.8.10
[cli][debug] Streamlink: 3.1.1
[cli][debug] Requests(2.27.1), Socks(1.6.7), Websocket(1.2.3)
[cli][debug] Arguments:
[cli][debug]  url=https://pluto.tv/live-tv/gunsmoke
[cli][debug]  stream=['best']
[cli][debug]  --loglevel=debug
[cli][info] Found matching plugin pluto for URL https://pluto.tv/live-tv/gunsmoke
[plugins.pluto][debug] slug=gunsmoke
[plugins.pluto][debug] app_version=5.107.0-4a2a6cc411c0f4fdbcce2a647a5add9995648102
[plugins.pluto][debug] path=/stitch/hls/channel/60f75771dfc72a00071fd0e0/master.m3u8
[utils.l10n][debug] Language code: en_CA
[cli][info] Available streams: 570k (worst), 1000k, 1500k, 2100k, 3100k (best)
[cli][info] Opening stream: 3100k (hls-pluto)
[cli][info] Starting player: /usr/bin/vlc
[stream.hls][debug] Reloading playlist
[cli][debug] Pre-buffering 8192 bytes
[stream.hls][debug] Segments in this playlist are encrypted
[stream.hls][debug] First Sequence: 1; Last Sequence: 5
[stream.hls][debug] Start offset: 0; Duration: None; Start Sequence: 3; End Sequence: None
[stream.hls][debug] Adding segment 3 to queue
[stream.hls][debug] Adding segment 4 to queue
[stream.hls][debug] Adding segment 5 to queue
[stream.hls][debug] Segment 3 complete
[cli.output][debug] Opening subprocess: /usr/bin/vlc --input-title-format https://pluto.tv/live-tv/gunsmoke -
[cli][debug] Writing stream to output
[stream.hls][debug] Segment 4 complete
[stream.hls][debug] Segment 5 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Segments in this playlist are encrypted
[stream.hls][debug] Adding segment 6 to queue
[stream.hls][debug] Adding segment 7 to queue
[stream.hls][debug] Segment 6 complete
[stream.hls][debug] Segment 7 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Segments in this playlist are encrypted
[stream.hls][debug] Adding segment 8 to queue
[stream.hls][debug] Segment 8 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Segments in this playlist are encrypted
[stream.hls][debug] Adding segment 9 to queue
[stream.hls][debug] Segment 9 complete
[cli][info] Player closed
[stream.segmented][debug] Closing worker thread
[stream.segmented][debug] Closing writer thread
[cli][info] Stream ended
[cli][info] Closing currently open stream...
greg@GregsAMC:~/Downloads$


greg@GregsAMC:/Downloads$ sudo -u nextpvr streamlink https://pluto.tv/live-tv/gunsmoke best -l debug
[sudo] password for greg:        
[cli][debug] OS:         Linux-5.4.0-962201191901-generic-x86_64-with-glibc2.29
[cli][debug] Python:     3.8.10
[cli][debug] Streamlink: 1.7.0
[cli][debug] Requests(2.22.0), Socks(1.6.7), Websocket(0.53.0)
error: No plugin can handle URL: https://pluto.tv/live-tv/gunsmoke
greg@GregsAMC:~/Downloads$ 


edited - removed tildes causing strikeout.

**Reply by mkbloke** [reply]
```
[cli][debug] Streamlink: 3.1.1
[cli][debug] Streamlink: 1.7.0
```
The problem is, you have two different versions of Streamlink installed.  I suspect the second listed one (1.7.0) is probably installed via a system package, so use your system package manager to find and remove it if that is the case.

I don't know how you installed the first listed version (3.1.1), perhaps locally for the user?  You can use the `which` command, i.e. `which streamlink` to find out where the `streamlink` command is in each case.

If you installed version 3.1.1 locally for a user, when you use `sudo` you might well need to provide the full path to the `streamlink` command for that user and also make sure the directory hierarchy has the correct permissions, such that the user you're switching to via `sudo` can gain access.

**Reply by lovatone** [reply]
Thank you very much for your help!
I removed 1.7.0 via software manager.

I ran the other commands but I'm lost on what to do next.
Here is the run, does it shed any light on my problem?

greg@GregsAMC:~/Downloads$ streamlink https://pluto.tv/live-tv/gunsmoke best -l debug
[cli][debug] OS:         Linux-5.4.0-962201191901-generic-x86_64-with-glibc2.29
[cli][debug] Python:     3.8.10
[cli][debug] Streamlink: 3.1.1
[cli][debug] Requests(2.27.1), Socks(1.6.7), Websocket(1.2.3)
[cli][debug] Arguments:
[cli][debug]  url=https://pluto.tv/live-tv/gunsmoke
[cli][debug]  stream=['best']
[cli][debug]  --loglevel=debug
[cli][info] Found matching plugin pluto for URL https://pluto.tv/live-tv/gunsmoke
[plugins.pluto][debug] slug=gunsmoke
[plugins.pluto][debug] app_version=5.107.0-4a2a6cc411c0f4fdbcce2a647a5add9995648102
[plugins.pluto][debug] path=/stitch/hls/channel/60f75771dfc72a00071fd0e0/master.m3u8
[utils.l10n][debug] Language code: en_CA
[cli][info] Available streams: 570k (worst), 1000k, 1500k, 2100k, 3100k (best)
[cli][info] Opening stream: 3100k (hls-pluto)
[cli][info] Starting player: /usr/bin/vlc
[stream.hls][debug] Reloading playlist
[cli][debug] Pre-buffering 8192 bytes
[stream.hls][debug] Segments in this playlist are encrypted
[stream.hls][debug] First Sequence: 0; Last Sequence: 4
[stream.hls][debug] Start offset: 0; Duration: None; Start Sequence: 2; End Sequence: None
[stream.hls][debug] Adding segment 2 to queue
[stream.hls][debug] Adding segment 3 to queue
[stream.hls][debug] Adding segment 4 to queue
[stream.hls][debug] Segment 2 complete
[cli.output][debug] Opening subprocess: /usr/bin/vlc --input-title-format https://pluto.tv/live-tv/gunsmoke -
[stream.hls][debug] Segment 3 complete
[cli][debug] Writing stream to output
[stream.hls][debug] Segment 4 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Segments in this playlist are encrypted
[stream.hls][debug] Adding segment 5 to queue
[stream.hls][debug] Segment 5 complete
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Segments in this playlist are encrypted
[stream.hls][debug] Adding segment 6 to queue
[stream.hls][debug] Segment 6 complete
[stream.hls][debug] Reloading playlist

greg@GregsAMC: /Downloads$ sudo -u nextpvr streamlink https://pluto.tv/live-tv/gunsmoke best -l debug
sudo: streamlink: command not found
greg@GregsAMC:~/Downloads$ 

greg@GregsAMC:$ which streamlink
/home/greg/.local/bin/streamlink
greg@GregsAMC: $ sudo which streamlink
[sudo] password for greg:        
greg@GregsAMC:~$

greg@GregsAMC: /Downloads$ echo $PATH
/home/greg/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/greg/.local/bin:/usr/share/streamlink/:/usr/share:/usr/share/doc:/home/greg/.local/bin/streamlink
greg@GregsAMC:~/Downloads$ 

greg@GregsAMC: /Downloads$ sudo echo $PATH
/home/greg/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/greg/.local/bin:/usr/share/streamlink/:/usr/share:/usr/share/doc:/home/greg/.local/bin/streamlink
greg@GregsAMC:~/Downloads$ 

greg@GregsAMC: /Downloads$ sudo bash
root@GregsAMC:/home/greg/Downloads# echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
root@GregsAMC:/home/greg/Downloads# 

edited : removed several tilde to prevent strikethrough

**Reply by mkbloke** [reply]
> /home/greg/.local/bin/streamlink

Right, so you can try: `sudo -Eu nextpvr streamlink https://pluto.tv/live-tv/gunsmoke best` (as bastimeyer has pointed out), but you might run into permissions issues (probably on `/home/greg`), which you will have to sort out appropriately.

If you need help understanding UNIX permissions you should probably seek it elsewhere though; as already mentioned by bastimeyer, this is not a Streamlink issue and is way off topic for discussion here, I'm afraid.

Good luck.

**Reply by lovatone** [reply]
Thank you for all your help!

**Comment by bastimeyer** [comment, marked as answer]
This is not a streamlink issue. Please read up on how to use sudo. Take a look at its man-page:

> `-E`, `--preserve-environment`

https://man.archlinux.org/man/sudo.8#E

Streamlink's CLI docs explicitly state where config and plugin files are loaded from, so make sure that you don't override these env vars:
- https://streamlink.github.io/cli.html#configuration-file
- https://streamlink.github.io/cli.html#plugin-specific-configuration-file
- https://streamlink.github.io/cli.html#sideloading-plugins

And when executing any kind of python programs, make sure that you are not breaking the python environment by overridng `PYTHONHOME`, `PYTHONPATH`, etc.

**Reply by lovatone** [reply]
Thanks,
The NextPvr support team gave me the two command lines to test.
i dont know much about sudo, or env vars, but ill give it a good try.
